test/test_encryption.py

```diff
@@ -2494,5 +2494,196 @@ def MongoClient(**kwargs):
         client_encryption.close()
 
 
+# https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#range-explicit-encryption
+class TestRangeQueryProse(EncryptionIntegrationTest):
+    @client_context.require_no_standalone
+    @client_context.require_version_min(6, 2, -1)
+    def setUp(self):
+        super().setUp()
+        self.key1_document = json_data("etc", "data", "keys", "key1-document.json")
+        self.key1_id = self.key1_document["_id"]
+        self.client.drop_database(self.db)
+        key_vault = create_key_vault(self.client.keyvault.datakeys, self.key1_document)
+        self.addCleanup(key_vault.drop)
+        self.key_vault_client = self.client
+        self.client_encryption = ClientEncryption(
+            {"local": {"key": LOCAL_MASTER_KEY}}, key_vault.full_name, self.key_vault_client, OPTS
+        )
+        self.addCleanup(self.client_encryption.close)
+        opts = AutoEncryptionOpts(
+            {"local": {"key": LOCAL_MASTER_KEY}},
+            key_vault.full_name,
+            bypass_query_analysis=True,
+        )
+        self.encrypted_client = rs_or_single_client(auto_encryption_opts=opts)
+        self.db = self.encrypted_client.db
+        self.addCleanup(self.encrypted_client.close)
+
+    def run_expression_find(self, name, expression, expected_elems, range_opts, use_expr=False):
+        find_payload = self.client_encryption.encrypt_expression(
+            expression=expression,
+            key_id=self.key1_id,
+            algorithm=Algorithm.RANGEPREVIEW,
+            query_type=QueryType.RANGEPREVIEW,
+            contention_factor=0,
+            range_opts=range_opts,
+        )
+        if use_expr:
+            find_payload = {"$expr": find_payload}
+        sorted_find = sorted(
+            self.encrypted_client.db.explicit_encryption.find(find_payload), key=lambda x: x["_id"]
+        )
+        for elem, expected in zip(sorted_find, expected_elems):
+            self.assertEqual(elem[f"encrypted{name}"], expected)
+
+    def run_test(self, name, range_opts, cast_func):
```

Comment:
I think "run_test_cases" would be a clearer name for this function.

Reply:
Done.